ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/validation/RequestValidator.java

```diff
@@ -137,15 +137,16 @@ public int validate(
     final StatementValidator<T> customValidator = (StatementValidator<T>)
         customValidators.get(statementClass);
 
+    final ConfiguredStatement<T> injected = injector.inject(configured);
     if (customValidator != null) {
-      customValidator.validate(configured, executionContext, serviceContext);
-    } else if (KsqlEngine.isExecutableStatement(configured.getStatement())) {
-      executionContext.execute(injector.inject(configured));
+      customValidator.validate(injected, executionContext, serviceContext);
+    } else if (KsqlEngine.isExecutableStatement(injected.getStatement())) {
+      executionContext.execute(injector.inject(injected));
```

Comment:
Why are we calling `inject` on a statement that has already had everything injected into it? If I'm understanding the injectors correctly, there should never be a case where it makes sense to call `inject` on a statement twice. Is that true?

Reply:
yes you are right... I think this was a refactor gone wrong.